overlord/configstate/configcore/netplan_test.go

```diff
@@ -0,0 +1,183 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore_test
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/dbusutil"
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/overlord/configstate/config"
+	"github.com/snapcore/snapd/overlord/configstate/configcore"
+	"github.com/snapcore/snapd/overlord/configstate/configcore/netplantest"
+	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type netplanSuite struct {
+	configcoreSuite
+	testutil.DBusTest
+
+	backend *netplantest.NetplanServer
+}
+
+var _ = Suite(&netplanSuite{})
+
+var mockNetplanConfigYaml = `
+network:
+  renderer: NetworkManager
+  version: 2
+`
+
+func (s *netplanSuite) SetUpTest(c *C) {
+	s.configcoreSuite.SetUpTest(c)
+	s.DBusTest.SetUpTest(c)
+
+	backend, err := netplantest.NewNetplanServer(mockNetplanConfigYaml)
+	c.Assert(err, IsNil)
+	s.AddCleanup(func() { c.Check(backend.Stop(), IsNil) })
+	s.backend = backend
+
+	// We mock the system bus with a private session bus, that is
+	// good enough for the unit tests. Spread tests cover the real
+	// bus.
+	restore := dbusutil.MockOnlySystemBusAvailable(s.SessionBus)
+	s.AddCleanup(restore)
+
+	restore = release.MockOnClassic(false)
+	s.AddCleanup(restore)
+
+	err = os.MkdirAll(filepath.Join(dirs.GlobalRootDir, "/etc/"), 0755)
+	c.Assert(err, IsNil)
+}
+
+func (s *netplanSuite) TearDownTest(c *C) {
+	s.configcoreSuite.TearDownTest(c)
+	s.DBusTest.TearDownTest(c)
+}
+
+func (s *netplanSuite) TestNetplanGetFromDbusNoSuchService(c *C) {
```

Comment:
nitpick^1000000: in other places you named a function using this capitalization: `DBus` :-)